src/olympia/devhub/views.py

```diff
@@ -1444,6 +1444,9 @@ def submit_addon_upload(request, channel):
 @dev_required(submitting=True)
 def submit_details(request, addon_id, addon):
     forms_, context = [], {}
+    if not addon.is_listed:
```

Comment:
😞 
move `latest_version = addon.find_latest_version(channel=amo.RELEASE_CHANNEL_LISTED)` up and `if not latest_version` then it's unlisted and we can redirect.